pkg/test/utils.go

```diff
@@ -60,3 +63,78 @@ func MustReadFile(t *testing.T, p string) []byte {
 	}
 	return b
 }
+
+// CompareGoldenFile performs a file comparison for a golden test.
+func CompareGoldenFile(t *testing.T, p string, got string, normalizers ...func(s string) string) {
+	writeGoldenOutput := os.Getenv("WRITE_GOLDEN_OUTPUT") != ""
+
+	for _, normalizer := range normalizers {
+		got = normalizer(got)
+	}
+
+	wantBytes, err := os.ReadFile(p)
+	if err != nil {
+		if writeGoldenOutput && os.IsNotExist(err) {
+			// expected when creating output for the first time
+		} else {
+			t.Fatalf("failed to read golden file %q: %v", p, err)
+		}
+	}
+	want := string(wantBytes)
```

Comment:
I might not familiar with golang string cast and I do not test the code locally. I'm just curious if `wantBytes` is nil(assuming golden file does not exist for the first time), is string(nil) a valid expression and what's the return value?

Reply:
https://go.dev/play/p/UJppOPhVWw5

Reply:
Cool, and the playground tool is awesome!

Reply:
Hmm correct me if I'm wrong, in your example, var b is not nil, it's an empty variable right? https://go.dev/play/p/DnUOhQ2aAuN

Reply:
But ... given revive is complaining about it also, I might as well make it more self-documenting :-)

Reply:
Thanks for the explanation! This might be where Golang is different from other languages. LGTM!